src/assets/CollectiblesController.test.ts

```diff
@@ -1373,4 +1373,242 @@ describe('CollectiblesController', () => {
       });
     });
   });
+
+  describe('findCollectibleByAddressAndTokenId', () => {
+    it('should return undefined when the collectible does not exist', async () => {
+      const mockTransactionId = '60d36710-b150-11ec-8a49-c377fbd05e27';
+
+      const collectible = {
+        address: '0x02',
+        tokenId: '1',
+        name: 'name',
+        image: 'image',
+        description: 'description',
+        standard: 'standard',
+        favorite: false,
+      };
+
+      collectiblesController.updateCollectibleByAddressAndTokenId(
+        collectible.address,
+        collectible.tokenId,
+        { ...collectible, transactionId: mockTransactionId },
+      );
+
+      expect(
+        collectiblesController.findCollectibleByAddressAndTokenId(
+          collectible.address,
+          collectible.tokenId,
+        ),
+      ).toBe(undefined);
+    });
+
+    it('should return the collectible by the address and tokenId with the prop transactionId', async () => {
+      const mockTransactionId = '60d36710-b150-11ec-8a49-c377fbd05e27';
+
+      const collectible = {
+        address: '0x02',
+        tokenId: '1',
+        name: 'name',
+        image: 'image',
+        description: 'description',
+        standard: 'standard',
+        favorite: false,
+      };
+
+      await collectiblesController.addCollectible(
+        collectible.address,
+        collectible.tokenId,
+        collectible,
+      );
+
+      collectiblesController.updateCollectibleByAddressAndTokenId(
+        collectible.address,
+        collectible.tokenId,
+        { ...collectible, transactionId: mockTransactionId },
+      );
+
+      expect(
+        collectiblesController.findCollectibleByAddressAndTokenId(
+          collectible.address,
+          collectible.tokenId,
+        ),
+      ).toStrictEqual({
+        address: '0x02',
+        description: 'description',
+        favorite: false,
+        image: 'image',
+        name: 'name',
+        standard: 'standard',
+        tokenId: '1',
+        transactionId: '60d36710-b150-11ec-8a49-c377fbd05e27',
+      });
```

Comment:
convert this object to a mockObject with transactionId , and move it to out of scope to other tests can use it